internal/mesh/internal/controllers/sidecarproxy/status/status.go

```diff
@@ -0,0 +1,74 @@
+package status
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/consul/proto-public/pbresource"
+)
+
+const (
+	StatusConditionDestinationAccepted = "DestinationAccepted"
```

Comment:
I feel like after we have all of the mesh group's controllers committed there's a pass we need to make to align our naming and error strategies here for status conditions since I think I've created some similar conditions to these, but did it slightly differently.

Reply:
Agreed. We might need to do a final pass once more things land.